src/get_target_data.py

```diff
@@ -0,0 +1,481 @@
+            logger.info(
+                f"No created_at field in modeled_clades metadata for {nowcast_string}. Defaulting to nowcast_date - 2 days."
+            )
+            tree_as_of = (nowcast_date - timedelta(days=2)).replace(
+                hour=23, minute=59, second=59, tzinfo=timezone.utc
+            )
+        else:
+            tree_as_of = datetime.fromisoformat(modeled_clades["meta"]["created_at"])
+
+    assignments = assign_clades(
+        nowcast_date,
+        sequence_as_of,
+        tree_as_of,
+        collection_min_date,
+        collection_max_date,
+    )
+    target_data = create_target_data(
+        assignments,
+        clade_list,
+        nowcast_string,
+        sequence_as_of.strftime("%Y-%m-%d"),
+        collection_min_date,
+        collection_max_date,
+    )
+
+    output_files = write_target_data(
+        nowcast_string,
+        sequence_as_of.strftime("%Y-%m-%d"),
+        target_data,
+        target_data_dir,
+    )
+
+    return output_files
+
+
+def assign_clades(
+    nowcast_date: datetime,
+    sequence_as_of: datetime,
+    tree_as_of: datetime,
+    collection_min_date: datetime,
+    collection_max_date: datetime,
+) -> Clade:
+    # Instantiate CladeTime object
+    ct = CladeTime(sequence_as_of=sequence_as_of, tree_as_of=tree_as_of)
+    logger.info(
+        {
+            "msg": "Starting clade assignment",
+            "CladeTime": ct,
+            "nowcast_date": nowcast_date.isoformat(timespec="seconds"),
+            "nextstrain_metadata_url": ct.url_sequence_metadata,
+            "collection_min_date": collection_min_date.isoformat(timespec="seconds"),
+            "collection_max_date": collection_max_date.isoformat(timespec="seconds"),
+        }
+    )
+
+    filtered_metadata = sequence.filter_metadata(
+        ct.sequence_metadata,
+        collection_min_date=collection_min_date,
+        collection_max_date=collection_max_date,
+    )
+
+    assignments = ct.assign_clades(filtered_metadata)
+    logger.info("Clade assignments complete")
+
+    return assignments
+
+
+def create_target_data(
+    assignments: Clade,
+    clade_list: list,
+    nowcast_string: str,
+    sequence_as_of_string: str,
+    collection_min_date: datetime,
+    collection_max_date: datetime,
+) -> tuple[pl.LazyFrame, pl.LazyFrame]:
+    """Return time series and oracle output target data."""
+    clade_list = [clade for clade in clade_list if clade != "other"]
+
+    time_series = assignments.summary.select(
+        ["location", "date", "clade_nextstrain", "count"]
+    ).filter(pl.col("clade_nextstrain").is_in(clade_list))
+
+    all_rows = (
+        pl.LazyFrame(
+            pl.date_range(
+                collection_min_date, collection_max_date, "1d", eager=True
+            ).alias("date")
+        )
+        .join(pl.Series("location", state_list).to_frame().lazy(), how="cross")
+        .join(pl.Series("clade_nextstrain", clade_list).to_frame().lazy(), how="cross")
```

Comment:
Noting that we're also going to want to keep the "other" group around here as well.